include/cglm/types.h

```diff
@@ -45,6 +46,30 @@ typedef CGLM_ALIGN_IF(32) vec4  mat4[4];
 typedef CGLM_ALIGN_IF(16) vec4  mat4[4];
 #endif
 
+// struct types
+typedef union CGLM_ALIGN_IF(16) vec3s {
+#ifndef CGLM_NO_ANONYMOUS_STRUCT
+  struct {
+    float x;
+    float y;
+    float z;
+  };
+#endif
+  vec3 raw;
+} vec3s;
+
+typedef union CGLM_ALIGN_IF(16) vec4s {
+#ifndef CGLM_NO_ANONYMOUS_STRUCT
+  struct {
+    float x;
+    float y;
+    float z;
+    float w;
+  };
+#endif
+  vec4 raw;
+} vec4s;
+
```

Comment:
And then `#include types-struct.h` in the `types.h`? Or should we make the user aware of what he is including and keep separated headers and the user include `types-struct.h` (that will also include `types.h` because of the `vecN` references)?

Reply:
> And then #include types-struct.h in the types.h?

No, I think struct api should be an alternative api, we should not mix them in library (users can do this if they want in their projects), and users could use:

- array api by including `cglm.h`
- struct api by including `structs.h` or `cglms.h` or `glms.h`...  this header could be like cglm.h which include all inline headers plus types-struct.h

`common.h` header should only keep common things. Also all headers must include `common.h` header in public headers except `types.h`. Please include `#include "cglm.h"` before include any headers :) 

Since we want to keep structs api separate, `types-struct.h` should not be included in `common.h` I think, but I'm not sure